app/src/main/java/cm/aptoide/pt/home/apps/AppsFragment.java

```diff
@@ -148,15 +154,39 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
   }
 
   @Override public void showUpdatesList(List<App> list) {
-    adapter.addUpdateAppsList(list);
+    if (list != null && !list.isEmpty()) {
+      adapter.addUpdateAppsList(list);
+    }
+    showUpdates = true;
+    if (canShowListOfApps() && recyclerView.getVisibility() != View.VISIBLE) {
+      recyclerView.scrollToPosition(0);
+      hideLoadingProgressBar();
+      recyclerView.setVisibility(View.VISIBLE);
+    }
```

Comment:
recyclerView.getVisibility != ...  can go inside canShowListOfApps
Entire if can be extracted to a method, since it's repeated for others